x/amm/types/calc_in_amt_given_out.go

```diff
@@ -68,6 +68,9 @@ func (p Pool) CalcInAmtGivenOut(
 	}
 
 	amountInWithoutSlippage := sdk.NewDecFromInt(tokenOut.Amount).Quo(rate)
+	if tokenAmountIn.IsZero(){
```

Comment:
So if it's zero, does it mean the user is not getting anything back? This seems to be an edge case for sure. We should still be able to fully close the position though.

Reply:
Would the tx be successful? We still need to make sure the position can be closed.

Reply:
@cryptokage1996 we should throw an error that says that the user should input a positive token amount in